server/item/recipe/recipe.go

```diff
@@ -0,0 +1,57 @@
+package recipe
+
+import "github.com/df-mc/dragonfly/server/item"
+
+// Recipe is implemented by all recipe types.
+type Recipe interface {
+	// Inputs returns the items required to craft the recipe.
+	Inputs() []InputItem
+	// UpdateInputs updates the inputs of the recipe.
+	UpdateInputs(inputs []InputItem)
```

Comment:
Is there any specific reason we allow this?

Reply:
Forgot why I did this, doesn't really make much sense, removed.